### PR TITLE
change additional warning suppress link flag to not override earlier link flags

### PR DESCRIPTION
**Pull request by orilevari**
**Description**: Change the ignore warnings link command to use target_link_options

**Motivation and Context**
The previous command was overriding the earlier set_target_properties LINK_FLAGS command

### CODE REVIEW

File: cmake/winml.cmake [hunk 1]
```diff
@@ -616,7 +616,5 @@ endif()
 # When cuda is enabled in the pipeline, it sets CMAKE_SHARED_LINKER_FLAGS which affects all targets including winml_dll.
 # However, there are no cuda imports in winml_dll, and the linker throws the 4199 warning.
 # This is needed to allow winml_dll build with cuda enabled.
-set_target_properties(winml_dll
-  PROPERTIES
-  LINK_FLAGS
-  "/ignore:4199")
\ No newline at end of file
+#set_property(TARGET winml_dll APPEND_STRING PROPERTY LINK_FLAGS " /ignore:4199")
```
Comment (emigsheil) [inline cmake/winml.cmake]: commented code?

Comment (ryanlai2) [inline cmake/winml.cmake]: Should this be removed?

**Comment by smk2007** [comment]
> 

dont we need to add dxcore here as well?

---
Refers to: cmake/winml.cmake:569 in d610ac9. [](commit_id = d610ac999ff94d8fcd12f1b5bf330191b4c81349, deletion_comment = False)

**Comment by smk2007** [comment]
> 

can you past the dump bin?

---
In reply to: [585480358](https://github.com/microsoft/onnxruntime/pull/3013#issuecomment-585480358) [](ancestors = 585480358)

---
Refers to: cmake/winml.cmake:569 in d610ac9. [](commit_id = d610ac999ff94d8fcd12f1b5bf330191b4c81349, deletion_comment = False)

**Comment by smk2007** [comment]
> 

From looking at the 19h1 redist, it looks like we determined that we needed to delayload all of these at that time, probably also need to delay load them now:

    set_target_properties(
        ${proj_name}
        PROPERTIES LINK_FLAGS
        "/DELAYLOAD:api-ms-win-core-winrt-error-l1-1-1.dll /DELAYLOAD:api-ms-win-core-com-l1-1-1.dll /DELAYLOAD:api-ms-win-core-libraryloader-l1-2-1.dll /DELAYLOAD:d3d12.dll /DELAYLOAD:d3d11.dll /DELAYLOAD:dxgi.dll /DELAYLOAD:directml.dll")


---
In reply to: [585480504](https://github.com/microsoft/onnxruntime/pull/3013#issuecomment-585480504) [](ancestors = 585480504,585480358)

---
Refers to: cmake/winml.cmake:569 in d610ac9. [](commit_id = d610ac999ff94d8fcd12f1b5bf330191b4c81349, deletion_comment = False)

File: cmake/winml.cmake (hunk 1 above)
Comment (tiagoshibata) [inline cmake/winml.cmake]: Remove commend

File: cmake/winml.cmake [hunk 2]
```diff
@@ -616,7 +616,5 @@ endif()
...
+target_link_options(winml_dll PRIVATE " /ignore:4199")
```
Comment (tiagoshibata) [inline cmake/winml.cmake]: This is adding `" /ignore:4199"` to the command line for me, no quotes is the correct way to do.

Comment (snnn) [reply cmake/winml.cmake]: "When cuda is enabled in the pipeline, it sets CMAKE_SHARED_LINKER_FLAGS which affects all targets including winml_dll." 

Then we can fix it, right? Or probably I already fix it?

**Comment by orilevari** [comment]
dumpbin after most recent commit: 
dumpbin /dependents windows.ai.machinelearning.dll
Microsoft (R) COFF/PE Dumper Version 14.24.28315.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file windows.ai.machinelearning.dll

File Type: DLL

  Image has the following dependencies:

    KERNEL32.dll
    api-ms-win-core-winrt-string-l1-1-0.dll
    api-ms-win-core-winrt-error-l1-1-0.dll
    ADVAPI32.dll
    OLE32.dll
    MSVCP140.dll
    VCRUNTIME140.dll
    api-ms-win-crt-runtime-l1-1-0.dll
    api-ms-win-crt-stdio-l1-1-0.dll
    api-ms-win-crt-math-l1-1-0.dll
    api-ms-win-crt-heap-l1-1-0.dll
    api-ms-win-crt-string-l1-1-0.dll
    onnxruntime.dll
    OLEAUT32.dll
    api-ms-win-core-winrt-l1-1-0.dll

  Image has the following delay load dependencies:

    DirectML.dll
    d3d12.dll
    d3d11.dll
    dxgi.dll
    DXCORE.dll
    api-ms-win-core-winrt-error-l1-1-1.dll

  Summary

        2000 .data
        1000 .didat
        C000 .pdata
       56000 .rdata
        2000 .reloc
        1000 .rsrc
       BA000 .text

**Comment by smk2007** [comment]
Yeah  there are a lot of references that are different from the 19h1 redist. Namely the dependency on the vcruntime140+msvcp140 are references to the public VC++ dlls. This will require that the caller has installed the vs redist, which we didnt require before...
 
We should not see any kernel32 or advapi32 dlls here either, because that would mean that the redist will not work on wcos devices...

---
In reply to: [585507608](https://github.com/microsoft/onnxruntime/pull/3013#issuecomment-585507608) [](ancestors = 585507608)

**Comment by smk2007** [comment]
Moreover this opens a larger can of worms for adobe, which has been using 19h1-redist without needing to install any other redists. I have not checked, but need to verify if onnxruntime.dll takes runtime dependencies on the vc redist.

---
In reply to: [585884673](https://github.com/microsoft/onnxruntime/pull/3013#issuecomment-585884673) [](ancestors = 585884673,585507608)

**Pull request closed without merging**